### PR TITLE
AArch64: Add "defined(TR_HOST_ARM64)" to AOT-related #ifdefs

### PR DESCRIPTION
**Pull request by knn-k**
This commit adds "defined(TR_HOST_ARM64)" to #ifdef lines that are
related to AOT.

Signed-off-by: KONNO Kazuhiro <konno@jp.ibm.com>

### CODE REVIEW

**Comment by knn-k** [comment]
I have two questions:

1. j9method.c contains `&& !defined(SMALL)` in a line I am changing.  Can I delete that part?
https://github.com/eclipse/openj9/blob/137181a7c6d44944a22a92058d38fd85820626f0/runtime/compiler/env/j9method.cpp#L1126

2. RelocationRuntime.hpp defines a symbol `TR_SHARED_CACHE_AOT_SE_PLATFORM` for some platforms, but the symbol is used from nowhere.  Can I delete the symbol definition?
https://github.com/eclipse/openj9/blob/137181a7c6d44944a22a92058d38fd85820626f0/runtime/compiler/runtime/RelocationRuntime.hpp#L28-L30

**Comment by 0xdaryl** [comment]
`SMALL` should never be defined anymore, so `!defined(SMALL)` should always evaluate to true.

I'll defer to @mpirvu or @dsouzai on whether `TR_SHARED_CACHE_AOT_SE_PLATFORM` can be deleted.

File: runtime/compiler/env/j9method.cpp [hunk 1]
```diff
@@ -1123,7 +1123,7 @@ static intptrj_t getInitialCountForMethod(TR_ResolvedMethod *rm, TR::Compilation
 
    intptrj_t initialCount = m->hasBackwardBranches() ? options->getInitialBCount() : options->getInitialCount();
 
-#if defined(J9VM_INTERP_AOT_COMPILE_SUPPORT) && defined(J9VM_OPT_SHARED_CLASSES) && (defined(TR_HOST_X86) || defined(TR_HOST_POWER) || defined(TR_HOST_S390) || defined(TR_HOST_ARM)) && !defined(SMALL)
```
Comment (knn-k) [inline runtime/compiler/env/j9method.cpp]: Deleted `&& !defined(SMALL)`.

File: runtime/compiler/runtime/RelocationRuntime.hpp [hunk 2]
```diff
@@ -25,10 +25,6 @@
 
 #include "j9cfg.h"
 
-#if defined(J9VM_INTERP_AOT_COMPILE_SUPPORT) && defined(J9VM_OPT_SHARED_CLASSES) && (defined(TR_HOST_X86) || defined(TR_HOST_POWER) || defined(TR_HOST_S390))
-   #define TR_SHARED_CACHE_AOT_SE_PLATFORM
```
Comment (knn-k) [inline runtime/compiler/runtime/RelocationRuntime.hpp]: Deleted the symbol definition.  I will revert it if the symbol is still used somewhere.

**Comment by knn-k** [comment]
Jenkins test sanity all jdk11

**Comment by 0xdaryl** [comment]
Jenkins test sanity all jdk11

**Pull request merged**